packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts

```diff
@@ -107,34 +107,23 @@ export class FuncPluginAdapter implements IDepsAdapter {
     return this.displayContinueWithLearnMoreLink(confirmMessage, dotnetManualInstallHelpLink);
   }
 
-  public async displayContinueWithLearnMoreLink(
-    message: string,
-    link: string
-  ): Promise<boolean> {
+  public async displayContinueWithLearnMoreLink(message: string, link: string): Promise<boolean> {
     if (!this._ctx.dialog) {
       // no dialog, always continue
       return true;
     }
-
-    const userSelected: string | undefined = (
-      await this._ctx.dialog.communicate(
-        new DialogMsg(DialogType.Ask, {
-          description: message,
-          type: QuestionType.Confirm,
-          options: [Messages.learnMoreButtonText, Messages.continueButtonText], // Cancel is added by default
-        })
-      )
-    ).getAnswer();
+    const res = await this._ctx.ui?.showMessage(
```

Comment:
Is cancel added by default?

Reply:
Yes, if use click cancel, undefined is returned.